test/test_cuda.py

```diff
@@ -803,6 +803,27 @@ def test_load_nonexistent_device(self):
         with self.assertRaisesRegex(RuntimeError, msg):
             _ = torch.load(buf)
 
+    def test_specify_improper_device_name(self):
+        import os
+        fname = "tempfile.pt"
+        try:
+            with self.assertRaisesRegex(RuntimeError, "Expected one of cpu"):
+                torch.save([torch.nn.Parameter(torch.randn(10, 10))], fname,
+                           _use_new_zipfile_serialization=True)
```

Comment:
nit: it's better if the `torch.save` invocation is outside of the `with` block, because it prevents you from incorrectly passing the test when `torch.save` throws and you never get to `torch.load`